### DISCUSSION TITLE
How can I use react query outside of react or other framework

### DISCUSSION THREAD

**Discussion by theaungmyatmoe** [TanStack/query#4771] [Q&A] [answered]
I wanna integrate react query in my **micro frontend** application, but this one is so stick with specific framework. I googled a lot to make sure react query is working with other framework without adapters.

What I want to do is that I wanna use react query as state **management over redux and fetching.**

But I want to use it in multiple framework **without worrying adapters**. Because when I sticks state with react provider how can I retrieve from from Vue? It is really complicated for me.

So, can I use react query as a framework agnostic query? Something like vallina query?

**If you know how to do it, please let me know.**

**Comment by TkDodo** [comment, marked as answer]
You can use the `@tanstack/query-core` package, which is in vanilla TypeScript and which is used by all adapters.

> Because when I sticks state with react provider how can I retrieve from from Vue? It is really complicated for me.

You can create one `QueryClient` and pass that to different Providers (vue, react). Then, they will use the same cache and share the data, so you can have something that was added to the cache with react and retrieve it with vue.

But honestly, I thought the goal of microfrontends is to isolate and separate. If two microfrontends write to the same cache and kinda depend on each other - that doesn't sound like a good idea to me ...

**Reply by turnerguo** [reply]
Where is the `@tanstack/query-core` docs? seems to be only framework-query docs there?

**Reply by TkDodo** [reply]
not sure what you're trying to do, but:

- https://tanstack.com/query/v4/docs/react/reference/QueryClient
- https://tanstack.com/query/v4/docs/react/reference/QueryCache
- https://tanstack.com/query/v4/docs/react/reference/MutationCache
- https://tanstack.com/query/v4/docs/react/reference/QueryObserver

**Reply by thianphat** [reply]
@TkDodo I'm also trying to figure out how to use Tanstack Query outside of react. I have a browser extension that I want to share cache between the popup and content script. Is there anyway to perhaps create a provider within the background script that holds all the cache data between the popup and ContentScript?

**Reply by TkDodo** [reply]
what do you mean "a provider" ? you have to create a `QueryClient` and then distribute that. Providers like the `<QueryClientProvider>` are react specific ...

**Reply by 0x100101** [reply]
> You can create one QueryClient and pass that to different Providers (vue, react). 

@TkDodo I'm looking in to doing this, but the types aren't compatible. For example, if I instantiate a new vanilla `QueryClient` from `@tanstack/query-core` and attempt to use it in Vue I'll get the following typescript error:

```
Property 'isRestoring' is missing in type 'import("@tanstack/query-core/build/lib/queryClient").QueryClient' but required in type 'import("@tanstack/vue-query/build/lib/queryClient").QueryClient'.
```

Thoughts on this?

**Reply by TkDodo** [reply]
Interesting. `vue-query` actually has its own query client that extends the one from the core:

https://github.com/TanStack/query/blob/01ee84997b3cf9fe39861bb31452c8f2ede8f00c/packages/vue-query/src/queryClient.ts#L31

That might be the issue? Other adapters don't do this so just to confirm, could you try it with either React, Solid or Svelte?

/cc @DamianOsipiuk

**Reply by ssijak** [reply]
@TkDodo how do you execute a mutation via QueryClient outside of a framework (no hooks, etc)?

**Reply by TkDodo** [reply]
```
queryClient
    .getMutationCache()
    .build(queryClient, options)
    .execute(variables)
```

**Reply by nemonemi** [reply]
I have noticed that one does not have to load the `QueryClient` from the `@tanstack/query-core` for this to work.

**Reply by czescwojtek** [reply]
Hey is there any documentation regarding query-core usage?

**Reply by sgcullen** [reply]
Are you able to point to any information/docs on @tanstack/query-core ?
Thanks so much!

**Reply by czescwojtek** [reply]
@sgcullen I don't think there are any. Jump into the source code to see how it's done
